spark3/src/test/java/org/apache/iceberg/spark/sql/TestTimestampWithoutZone.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.base.Joiner;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.spark.SparkCatalogTestBase;
+import org.apache.iceberg.spark.SparkSessionCatalog;
+import org.apache.iceberg.spark.SparkUtil;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+import org.joda.time.DateTime;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class TestTimestampWithoutZone extends SparkCatalogTestBase {
+
+  private static final String newTableName = "created_table";
+  private final Map<String, String> config;
+
+  private static final Schema schema = new Schema(
+          Types.NestedField.required(1, "id", Types.LongType.get()),
+          Types.NestedField.required(2, "ts", Types.TimestampType.withoutZone()),
+          Types.NestedField.required(3, "tsz", Types.TimestampType.withZone())
+  );
+
+  private final List<Object[]> values = ImmutableList.of(
+          row(1L, toTimestamp("2021-01-01T00:00:00.0"), toTimestamp("2021-02-01T00:00:00.0")),
+          row(2L, toTimestamp("2021-01-01T00:00:00.0"), toTimestamp("2021-02-01T00:00:00.0")),
+          row(3L, toTimestamp("2021-01-01T00:00:00.0"), toTimestamp("2021-02-01T00:00:00.0"))
+  );
+
+  @Parameterized.Parameters(name = "catalogName = {0}, implementation = {1}, config = {2}")
+  public static Object[][] parameters() {
+    return new Object[][]{{"spark_catalog",
+            SparkSessionCatalog.class.getName(),
+            ImmutableMap.of(
+                    "type", "hive",
+                    "default-namespace", "default",
+                    "parquet-enabled", "true",
+                    "cache-enabled", "false"
+            )}
+    };
+  }
+
+  public TestTimestampWithoutZone(String catalogName, String implementation, Map<String, String> config) {
+    super(catalogName, implementation, config);
+    this.config = config;
+  }
+
+  @Before
+  public void createTables() {
+    spark.conf().set(SparkUtil.HANDLE_TIMESTAMP_WITHOUT_TIMEZONE, "false");
+    validationCatalog.createTable(tableIdent, schema);
+  }
+
+  @After
+  public void removeTables() {
+    validationCatalog.dropTable(tableIdent, true);
+    sql("DROP TABLE IF EXISTS %s", newTableName);
+  }
+
+  @Test
+  public void testWriteTimestampWithoutZoneError() {
+    String errorMessage = String.format("Write operation performed on a timestamp without timezone field while " +
+            "'%s' set to false should throw exception", SparkUtil.HANDLE_TIMESTAMP_WITHOUT_TIMEZONE);
+    Runnable insert = () -> sql("INSERT INTO %s VALUES %s", tableName, rowToSqlValues(values));
```

Comment:
You can just put this right into the AssertThrows, we usually do this.

```
AssertThrows(errorMessage, IllegalArguementException.class, SparkUtil.TIMESTAMP_WITHOUT_TIMEZONE_ERROR,
  () -> code);
```

Reply:
Instead of using the rowToSqlValues function, it may be simpler to just do a spark.createDataset(values).write command. That way we don't have to worry about string building

Reply:
I have moved insert Runnable and errorMessage to the AssertThrows 

I would like to leave `rowToSqlValues` if you do not mind. From my point of view this is better to have pure sql in this test